src/zilliandomizer/options/__init__.py

```diff
@@ -105,6 +105,9 @@ class Options:
     - rooms: random terrain inside rooms, but path through base is vanilla
     - full: random path through base
     """
+    map_gen_seed: Union[int, str] = "-1"
```

Comment:
Why is this type `Union[int, str]`?
I'm thinking just `int` would be simpler.

You have a `options.map_gen_seed > -1` which will crash if it's `str`
and some other places make it look like it should only be `int` also.